src/trace/listener.ts

```diff
@@ -131,13 +131,18 @@ export class TraceListener {
 
     if (this.triggerTags) {
       const statusCode = extractHTTPStatusCodeTag(this.triggerTags, result);
+
       // Store the status tag in the listener to send to Xray on invocation completion
       this.triggerTags["http.status_code"] = statusCode!;
       if (this.tracerWrapper.currentSpan) {
         this.tracerWrapper.currentSpan.setTag("http.status_code", statusCode);
       }
       if (this.inferredSpan) {
         this.inferredSpan.setTag("http.status_code", statusCode);
+
+        if(statusCode) {
+          this.inferredSpan.setTag("error", Number(statusCode) >= 500);
```

Comment:
@astuyve Is there a better way to assign the error from `currentSpan` to `inferredSpan` ?

Here is how the traces looks without those changes:
<img width="1004" alt="Screen Shot 2022-03-01 at 11 44 35 AM" src="https://user-images.githubusercontent.com/1880766/156145189-731abb1f-3126-4a84-8c5f-6ff484f89249.png">

As you can see the error information is not forwarded to the API Gateway inferred span

Reply:
Got it, can you please help me understand how I can actually get the error from the `wrappedCurrentSpan` ? I don't think that's a really good idea to actually set the error tag based on the lambda status code.